sourcegraph/cody
# Send client-name, client-version query string parameters to completions

Closes CODY-1300

Sourcegraph 5.4 or later completions endpoint will start to check a minimum version asserted by the client. This starts sending the relevant header.

Part of sourcegraph/sourcegraph#62116

## Test plan

CI

Note updated recordings have client-name, client-version query string parameters.

### Manual test

Sign in to sg02.sourcegraphcloud.com and do a chat. You should see an error message like "Request Failed: ClientCodyIgnoreCompatibilityError: Cody for vscode version "1.16.0" doesn't match version constraint ">= 1.20.0" Note, the "vscode" and "1.16.0" strings come from the client.

![Screenshot 2024-05-02 at 19 26 28](https://github.com/sourcegraph/cody/assets/55120/0f167752-cfd4-42a3-8040-050bf5925479)

Generate an autocomplete with alt-\ and check the Cody by Sourcegraph output category contains a cody.completion/error log:

```
    "privateMetadata": {
      "message": "Request to https://sg02.sourcegraphcloud.com/.api/completions/code?client-name=vscode&client-version=1.16.0 failed with 406 Not Acceptable: ClientCodyIgnoreCompatibilityError: Cody for vscode version \"1.16.0\" doesn't match version constraint \">= 1.20.0\"\n"
    }
```

JetBrains: Run JetBrains plugin with this agent and do similar actions. Error messages indicate the right client name and version are sent:

![Screenshot 2024-05-02 at 19 38 58](https://github.com/sourcegraph/cody/assets/55120/619217ba-e55d-4d4e-9dd8-f94f09ab77f8)
